Tests/SentryTests/Protocol/SentryMessageTests.swift

```diff
@@ -0,0 +1,54 @@
+import XCTest
+
+class SentryMessageTests: XCTestCase {
+    
+    private class Fixture {
+        let stringMaxCount = 8_192
+        let maximumCount: String
+        let tooLong: String
+        
+        init() {
+            maximumCount = String(repeating: "a", count: stringMaxCount)
+            tooLong = String(repeating: "a", count: stringMaxCount + 1)
+        }
+    }
+    
+    private let fixture = Fixture()
+    
+    func testTruncateFormatted() {
+        let message = SentryMessage(formatted: "aaaaa")
+        XCTAssertEqual(5, message.formatted?.count)
+        
+        message.formatted = fixture.maximumCount
+        XCTAssertEqual(fixture.stringMaxCount, message.formatted?.count)
+        
+        message.formatted = fixture.tooLong
+        XCTAssertEqual(fixture.stringMaxCount, message.formatted?.count)
+    }
+    
+    func testTruncateMessage() {
+        let message = SentryMessage()
+        message.message = "aaaaa"
+        
+        XCTAssertEqual(5, message.message?.count)
+        
+        message.message = fixture.maximumCount
+        XCTAssertEqual(fixture.stringMaxCount, message.message?.count)
+        
+        message.message = fixture.tooLong
+        XCTAssertEqual(fixture.stringMaxCount, message.message?.count)
+    }
+    
+    func testSerialize() {
+        let message = SentryMessage()
+        message.formatted = "Something %s"
+        message.message = "A message"
+        message.params = ["my", "params"]
+        
+        let actual = message.serialize()
+        
+        XCTAssertEqual(message.formatted, actual["formatted"] as? String)
+        XCTAssertEqual(message.message, actual["message"] as? String)
+        XCTAssertEqual(message.params, actual["params"] as? [String])
+    }
```

Comment:
H: not really related to this test but think about the use case:
the device has a cached event with the old message field called `message` and its a `String`, the new version of the App has the upgraded SDK and now `message` is a type, how would the serialization behave? does it crash? does it callback and set the string to the formatted field? does it fail and delete the event altogether?

Reply:
I like it a lot that you think about such edge cases 👏 . Until now we only serialize an event and don't deserialize it. So we just store it as an envelope to disk and never look into the envelope item containing the event. Furthermore, we just remove events cached to [disk before version 6.x](https://github.com/getsentry/sentry-cocoa/blob/master/Sources/Sentry/SentryFileManager.m#L64-L66). So we are good.

Reply:
I'm going to add the information to the migration guide tough, that we just delete cached events before v 6.x.

Reply:
nice, thanks